app/models/claim/matching_attribute_finder.rb

```diff
@@ -15,17 +13,29 @@ def initialize(source_claim, claims_to_compare = Claim.submitted)
     end
 
     def matching_claims
-      predicates = []
-      values = []
-      ATTRIBUTES_TO_MATCH.each do |attribute|
-        value = @source_claim.read_attribute(attribute)
-        next if value.blank?
-
-        predicates << "LOWER(#{attribute}) = LOWER(?)"
-        values << value
+      claims = @claims_to_compare.where.not(id: @source_claim.id)
+      match_queries = nil
+
+      ATTRIBUTE_GROUPS_TO_MATCH.each do |attributes|
+        vals = values_for_attributes(attributes)
```

Comment:
Unless there's a compelling reason to abbreviate, full variable names are generally better as they are easier to read and understand.